addons/event_booth_sale/models/sale_order_template.py

```diff
@@ -0,0 +1,8 @@
+from odoo import fields, models
+
+class SaleOrderTemplateLine(models.Model):
+    _inherit = "sale.order.template.line"
+
+    product_id = fields.Many2one(
+        'product.product', 'Product', check_company=True,
+        domain=[('sale_ok', '=', True), ('detailed_type', '!=', 'event'), ('detailed_type', '!=', 'event_booth')])
```

Comment:
```suggestion
    product_id = fields.Many2one(
        domain=[('sale_ok', '=', True), ('detailed_type', '!=', 'event'), ('detailed_type', '!=', 'event_booth')])
```
Do not specify again the attributes where the value doesn't change, it's forwarded automatically by the orm.